modules/system_monitor.py

```diff
@@ -226,9 +226,20 @@ def _extract_task_id(*args):
         return uuid.uuid4().hex
 
 
+def _get_system_monitor_config(request: gr.Request):
+    headers = dict(request.headers)
```

Comment:
You don't need to transfer `request.headers` to dict. `request.headers` follows the Mapping protocol.
Also you don't need to use two keys (`x-diffus-system-monitor-url` and `X-Diffus-System-Monitor-Url`) to get the specific value, `Headers.get` will handle it for you.

Just
```python
system_monitor_api_secret = headers.get('x-diffus-system-monitor-api-secret', '')
```
is OK